plugins/attachment-resources/src/components/AttachmentPresenter.svelte

```diff
@@ -106,49 +110,41 @@
 {:else}
   <div class="flex-row-center attachment-container">
     {#if value}
-      {#await getBlobRef(value.file, value.name, sizeToWidth('large')) then valueRef}
-        <a
-          class="no-line"
-          style:flex-shrink={0}
-          href={valueRef.src}
-          download={value.name}
-          on:click={clickHandler}
-          on:mousedown={middleClickHandler}
-          on:dragstart={dragStart}
-        >
-          {#if showPreview && isImage(value.type)}
-            <img
-              src={valueRef.src}
-              data-id={value.file}
-              srcset={valueRef.srcset}
-              class="flex-center icon"
-              class:svg={value.type === 'image/svg+xml'}
-              class:image={isImage(value.type)}
-              alt={value.name}
-            />
-          {:else}
-            <div class="flex-center icon">
-              {iconLabel(value.name)}
-            </div>
-          {/if}
-        </a>
-        <div class="flex-col info-container">
-          <div class="name">
-            <a href={valueRef.src} download={value.name} on:click={clickHandler} on:mousedown={middleClickHandler}>
-              {trimFilename(value.name)}
-            </a>
+      {#if canLinkPreview}
+        {#await getJsonOrEmpty(value.file, value.name)}
+          <Spinner size="small" />
+        {:then linkPreviewDetails}
+          <div class="flex-center icon">
+            {#if linkPreviewDetails.icon !== undefined && !useDefaultIcon}
+              <img
+                src={linkPreviewDetails.icon}
+                class="link-preview-icon"
+                alt="link-preview"
+                on:error={() => {
+                  useDefaultIcon = true
+                }}
+              />
+            {:else}
+              <svg class="link-preview-icon" viewBox="0 0 28 28" fill="none" xmlns="http://www.w3.org/2000/svg">
```

Comment:
nit: please  extract this into separate component

Reply:
thanks, fixed.